media/js/firefox/hello/start.js

```diff
@@ -0,0 +1,364 @@
+            _getText('sharedText')
+        );
+    }
+
+    // register for Hello UITour events
+    function bindHelloEvents() {
+        Mozilla.UITour.observe(function(event, data) {
+
+            switch(event) {
+            case 'Loop:ChatWindowOpened':
+                // only show invite step if conversation is not waiting
+                if (tourStep === 'get-started') {
+
+                    tourStep = 'invite';
+                    showTourStep();
+
+                    // User has interacted with FTE, so tell Firefox to resume
+                    // when they have their first conversation.
+                    resumeTourOnFirstJoin();
+
+                    // track user has clicked "Start a conversation" button
+                    gaTrack(['_trackEvent', '/hello/start interactions', 'tour', 'StartConversation-Tour']);
+
+                } if (tourStep === 'invite') {
+
+                    showTourStep();
+                } else if (tourStep === 'conversation-waiting') {
+
+                    tourStep = 'conversation-open';
+                    showTourStep();
+                    // set param to done, so if the user shares the URL
+                    // the next person can still take the tour from the start.
+                    replaceURLState('waiting', 'done');
+                    // track conversation connect
+                    trackGAConversationConnect();
+                }
+                break;
+            case 'Loop:ChatWindowClosed':
+                hideUITourHighlights();
+                break;
+            case 'Loop:ChatWindowHidden':
+                hideUITourHighlights();
+                break;
+            case 'Loop:ChatWindowDetached':
+                hideUITourHighlights();
+                break;
+            case 'Loop:IncomingConversation':
+                if (data && data.conversationOpen === true) {
+                    tourStep = 'conversation-open';
+
+                    hideUITourHighlights();
+                    showPageState('end', true);
+
+                    // track conversation connect
+                    trackGAConversationConnect();
+
+                } else if (data && data.conversationOpen === false) {
+                    tourStep = 'conversation-waiting';
+                    showHelloPanel(highlightRoomList);
+                }
+                break;
+            case 'Loop:RoomURLCopied':
+                tourStep = 'shared';
+                showTourStep();
+                // track user has clicked copy button
+                gaTrack(['_trackEvent', '/hello/start interactions', 'tour', 'URLCopied-Tour']);
+                break;
+            case 'Loop:RoomURLEmailed':
+                tourStep = 'shared';
+                showTourStep();
+                // track user has clicked email button
+                gaTrack(['_trackEvent', '/hello/start interactions', 'tour', 'URLEmailed-Tour']);
+            }
+
+        }, function () {
```

Comment:
Even though we're not doing anything with this callback, I'd actually like to keep it in here as a reminder to assist debugging locally in the future.